mne/preprocessing/maxwell.py

```diff
@@ -90,25 +103,55 @@ def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3,
 
     # Magnetometers (with coil_class == 1.0) must be scaled by 100 to improve
     # numerical stability as they have different scales than gradiometers
-    coil_scale = np.ones(len(picks))
-    coil_scale[np.array([coil['coil_class'] == 1.0 for coil in coils])] = 100.
+    magscale = 100.
+    coil_scale = np.ones((len(picks), 1))
+    coil_scale[np.array([coil['coil_class'] == 1.0
```

Comment:
you shouln't need to do the `np.array(...)` here, a list should be fine (`numpy` should automatically cast to array, I think)